x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/rules_settings_modal.tsx

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useState } from 'react';
+import { RulesSettingsFlappingProperties } from '@kbn/alerting-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiHorizontalRule,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiSwitch,
+  EuiSwitchProps,
+  EuiPanel,
+  EuiText,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import { useKibana } from '../../../common/lib/kibana';
+import {
+  RulesSettingsFlappingFormSection,
+  RulesSettingsFlappingFormSectionProps,
+  RulesSettingsFlappingTitle,
+} from './rules_settings_flapping_form_section';
+import { useGetFlappingSettings } from '../../hooks/use_get_flapping_settings';
+import { useUpdateFlappingSettings } from '../../hooks/use_update_flapping_settings';
+import { CenterJustifiedSpinner } from '../center_justified_spinner';
+
+const flappingDescription = i18n.translate(
+  'xpack.triggersActionsUI.rulesSettings.modal.flappingDetectionDescription',
+  {
+    defaultMessage:
+      'Alerts that go quickly go between active and recovered are considered flapping. Detecting these changes and minimizing new alert generation can help reduce unwanted noise in your alerting system.',
+  }
+);
+
+const flappingEnableLabel = i18n.translate(
+  'xpack.triggersActionsUI.rulesSettings.modal.enableFlappingLabel',
+  {
+    defaultMessage: 'Enabled flapping detection (recommended)',
```

Comment:
In the Advanced Settings UI, it uses "on" and "off" instead of "enabled" and "disabled". Can we do the same here (i.e. the text changes from "On" to "Off" depending on the state of the toggle)?: 
```suggestion
    defaultMessage: 'On (recommended)',
```